src/main/java/org/opensearch/security/configuration/SecurityFlsDlsIndexSearcherWrapper.java

```diff
@@ -116,6 +127,37 @@ protected DirectoryReader dlsFlsWrap(final DirectoryReader reader, boolean isAdm
             log.trace("dlsFlsWrap(); index: {}; privilegeEvaluationContext: {}", index.getName(), privilegesEvaluationContext);
         }
 
+        String indexName = shardId != null ? shardId.getIndexName() : null;
+        Set<String> resourceIds;
+        if (this.isResourceSharingEnabled
+            && !Strings.isNullOrEmpty(indexName)
+            && OpenSearchSecurityPlugin.getResourceIndices().contains(indexName)) {
+            resourceIds = this.resourceAccessHandler.getAccessibleResourceIdsForCurrentUser(indexName);
```

Comment:
Under the hood, this makes a sync index operation. 

We are here in an extremely low level operation. I am pretty sure that this will not work, even if it would be an async operation (which it cannot be due to the API constraints).

Reply:
Could you explain a bit more in detail?